### PR TITLE
Add support for GitHub Enterprise in CatalogImporterClient

### PR DESCRIPTION
**Pull request by O5ten**
Uses the configApi to pick up `integrations.github.host` and `integrations.github.apiBaseUrl` and read those with `readGitHubIntegrationConfig` and pass it on to the `CatalogImporterClient`. 

Found no tests to boot so i added none because i'm unsure how to test it. Tested against our own GHE where it works fine. Relying on that the first integration config is the one to be used. The readGitHubIntegrationConfig seems to default to adding a ref to github.com if none is provided. 

image

image

### CODE REVIEW

File: plugins/catalog-import/src/api/CatalogImportApi.ts
Comment (Rugvip) [inline plugins/catalog-import/src/api/CatalogImportApi.ts:19]: Just had a discussion the other day that this should be a private package that isn't used directly 😅 . The intention is really to provide an API from the app that can be used for SCM access rather than depending on this package directly. There are some dependencies like token generation features and default scm providers that only the app/backend can decide, so we want this logic injected instead.

File: plugins/catalog-import/src/util/useGithubRepos.ts
Comment (Rugvip) [inline plugins/catalog-import/src/util/useGithubRepos.ts:32]: We actually have the location of the repo right? So can match on the hostname instead of always selecting the first one.

**Review by Rugvip — COMMENTED** [review]
Thanks! 😁 

As mentioned in the comment we likely don't want `@backstage/integration` to be used directly because it impossible for the app to plug in custom logic. Right now it's really the only sane way to do this though so I think we could go ahead with this, but please slap a big `TODO` on it that we're intending to refactor the direct use of the package.

The actual nit I think is best to fix before merging is to select the config based on the `selectedRepo.location` instead of always picking the first one.

File: plugins/catalog-import/src/util/useGithubRepos.ts
Comment (freben) [reply plugins/catalog-import/src/util/useGithubRepos.ts:32]: Next question is, are these config values even exposed in the frontend? Haha, I haven't checked.

**Comment by freben** [comment]
Yep newly introduced stuff gets picked up so fast... :)

Yeah indeed this should have been added as a (not yet created...) core api instead of direct import, got to keep an eye on replacing this if we merge.

**Comment by adamdmharvey** [comment]
This should close my issue https://github.com/backstage/backstage/issues/3541 I believe 🎉 thx @O5ten

File: plugins/catalog-import/src/util/useGithubRepos.ts
Comment (O5ten) [reply plugins/catalog-import/src/util/useGithubRepos.ts:32]: @freben They worked for me! So i guess so!

File: plugins/catalog-import/src/api/CatalogImportApi.ts
Comment (O5ten) [reply plugins/catalog-import/src/api/CatalogImportApi.ts:19]: Wouldn't that be the same for the github.com support here only we chose to hardcode it instead of using the configuration. I'd argue that the shortcut was taken before this PR. :)

**Comment by O5ten** [comment]
I'll add that `//TODO tomorrow` and select the provide based on the selectedRepo. :)

File: plugins/catalog-import/src/util/useGithubRepos.ts
Comment (O5ten) [reply plugins/catalog-import/src/util/useGithubRepos.ts]: Resolved in the latest commit, used the following: 
```ts
const [hostname, ownerName, repoName] = selectedRepo.location.split('/').slice(-3);
...
const githubIntegrationConfig = configs.find(v => v.host === hostname);
```

**Comment by O5ten** [comment]
Resolved comments on //TODO and config-retrieval.

File: plugins/catalog-import/src/util/useGithubRepos.ts
Comment (freben) [reply plugins/catalog-import/src/util/useGithubRepos.ts]: Other parts of the repo is gathering around `git-url-parse`, search around the repo. This would be something like

```ts
import parseGitUrl from 'git-url-parse';
  const {
    name: repoName,
    owner: ownerName,
    resource: host,
  } = parseGitUrl(url);
```

Comment (O5ten) [reply plugins/catalog-import/src/util/useGithubRepos.ts]: Looks a LOT better to me to use that. Thought the split-splice was a little iffy, but hey, i'm the guest here so i'll conform to what i find. :)

Comment (O5ten) [reply plugins/catalog-import/src/util/useGithubRepos.ts]: @freben Now i'm also using git-url-parse. Worked like a charm!

File: plugins/catalog-import/src/util/useGithubRepos.ts
Comment (freben) [inline plugins/catalog-import/src/util/useGithubRepos.ts:33]: @O5ten you still have this here instead of git-url-parse - maybe you forgot to push the change?

**Review by freben — APPROVED** [review]

**Comment by freben** [comment]
So uh, now there's a merge conflict :) But so close!

File: plugins/catalog-import/package.json
Comment (freben) [inline plugins/catalog-import/package.json:28]: this is now 0.1.3 after the release went out

**Comment by freben** [comment]
Excellent. I'll merge once it's green.

**Pull request merged**
